src/platform/qt/library/LibraryController.cpp

```diff
@@ -0,0 +1,173 @@
+/* Copyright (c) 2014-2017 waddlesplash
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+#include "LibraryController.h"
+
+#include "../GBAApp.h"
+#include "LibraryGrid.h"
+#include "LibraryTree.h"
+
+namespace QGBA {
+
+void AbstractGameList::addGames(QList<LibraryEntry*> items) {
+	for (LibraryEntry* o : items) {
+		addGame(o);
+	}
+}
+void AbstractGameList::removeGames(QList<LibraryEntry*> items) {
+	for (LibraryEntry* o : items) {
+		removeGame(o);
+	}
+}
+
+LibraryController::LibraryController(QWidget* parent_widget, const QString& path, ConfigController* config)
```

Comment:
Because this class (just like the LibraryView) are used inside a .ui file, and uic passes `parent` as the first argument to the constructor.